homeassistant/components/doorbird.py

```diff
@@ -59,27 +65,105 @@ def setup(hass, config):
                       device_ip, str(status[1]))
         return False
 
-    if config[DOMAIN].get(CONF_DOORBELL_EVENTS):
+    # Get the URL of this server
+    hass_url = hass.config.api.base_url
+
+    # Override it if another is specified in the component configuration
+    if config[DOMAIN].get(CONF_CUSTOM_URL):
+        hass_url = config[DOMAIN].get(CONF_CUSTOM_URL)
+        _LOGGER.info("DoorBird will connect to this instance via %s",
+                     hass_url)
+
+    # Register for push notifications if enabled
+    if config[DOMAIN].get(CONF_DOORBELL_EVENTS)\
+            or config[DOMAIN].get(CONF_MOTION_EVENTS):
         # Provide an endpoint for the device to call to trigger events
         hass.http.register_view(DoorbirdRequestView())
 
-        # Get the URL of this server
-        hass_url = hass.config.api.base_url
+        # Create a new schedule (24/7)
+        schedule = DoorBirdScheduleEntrySchedule()
+        schedule.add_weekday(0, 604800)
+
+        # Register doorbell push schedule if enabled
+        if config[DOMAIN].get(CONF_DOORBELL_EVENTS):
+            doorbell_num = config[DOMAIN].get(CONF_DOORBELL_NUM, 1)
+            register_sensor(hass_url, device, SENSOR_DOORBELL, schedule,
+                            doorbell_num)
 
-        # Override it if another is specified in the component configuration
-        if config[DOMAIN].get(CONF_CUSTOM_URL):
-            hass_url = config[DOMAIN].get(CONF_CUSTOM_URL)
-            _LOGGER.info("DoorBird will connect to this instance via %s",
-                         hass_url)
+        # Register motion push schedule if enabled
+        if config[DOMAIN].get(CONF_MOTION_EVENTS):
+            register_sensor(hass_url, device, SENSOR_MOTION, schedule)
 
-        # This will make HA the only service that gets doorbell events
-        url = '{}{}/{}'.format(hass_url, API_URL, SENSOR_DOORBELL)
-        device.reset_notifications()
-        device.subscribe_notification(SENSOR_DOORBELL, url)
+    # Deregister doorbell pushes if not enabled
```

Comment:
Shouldn't this be done at each home assistant stop instead?

Reply:
Yes, but shouldn't we instead deregister everything at stop? Otherwise we need to start again to deregister.

Reply:
Where are we registering the notifications?

Reply:
When HA starts, it will register (with the DoorBird device itself) that it wants doorbell and/or motion notifications, depending on which are enabled in the HA config. It will also check to see if it there are old records (used to be in config but then got removed/commented out) for HA that it doesn't want anymore and remove them, otherwise putting a notification in your HA config is permanent even if you remove it later.

We *could* deregister at stop, but right now I just leave them there so they don't need to be re-registered next time HA restarts. That way there aren't any EEPROM writes on the device every time HA restarts, only when the config changes.

Reply:
Sounds good!